### ISSUE TITLE
numpy>=2.0.0 compatibility

### ISSUE THREAD

**Issue by martingabelmann** [gplepage/vegas#36]
Hi,

first:  thanks for this great piece of software!

Upgrading to numpy 2.0.0 I get the following error message upon import:
`ValueError: numpy.dtype size changed, may indicate binary incompatibility. Expected 96 from C header, got 88 from PyObject`

I'm relatively sure that it has to do with numpy2 since I can also reproduce it in a clean docker container:
```
## docker run -ti --rm python sh       
# pip install -q vegas
# python -c "import vegas"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/usr/local/lib/python3.12/site-packages/vegas/__init__.py", line 60, in <module>
    from ._vegas import RAvg, RAvgArray, RAvgDict
  File "src/vegas/_vegas.pyx", line 1, in init vegas._vegas
ValueError: numpy.dtype size changed, may indicate binary incompatibility. Expected 96 from C header, got 88 from PyObject
```

I also tried to fix it by first installing numpy2 and then build vegas from source - no luck. I guess it has to do with the API changes in numpy?

Using `numpy==1.26.4` in vegas' setup.cfg fixed the issue for me but for people on numpy>2 doing `pip install vegas` is broken...

**Comment by gplepage** [comment]
Thanks for the alert. I will have a new version up in a few days that will fix this.

**Comment by gplepage** [comment]
This should be fixed in version 6.1.3.

**Comment by martingabelmann** [comment]
Thanks a lot for the fix!
